### ISSUE TITLE
Allow checking out the commits when reviewing

### ISSUE THREAD

**Issue by PeterBenc** [microsoft/vscode-pull-request-github#3301]
If possible, I like to review PRs commits by commit. I d like to be able to check out these commits one by one, see what has changed and be able to comment on it and be able to edit them and try out some suggestions before proposing them.

Current behavior: 
 - commenting on separate commits doesn't work, (already opened an issue for that)
 - can not make changes to the files
 - without checking them out I cant see typescript types on hover etc.
 
 Propose behavior: 
 - allow checking out the commit
 - this would allow the files to be edited and have all vs code features

**Comment by ELLIOTTCABLE** [comment]
Yeah — a lot of VScode features aren't possible unless the buffers exist 'on-disk.' Such virtual files don't have LSP support from most LSPs, etceteras.

These LSP features work in the "files" view, because the left-side of the diff *is* the on-disk file; but they don't work in the "commits" view, because both sides of the diff are virtual.

Adding a button to the right edge of each commit-row in the "Commits" view to checkout that commit, thus allowing all of the diffs to be on-disk, would be fantastic. (This is somewhat useless until #2793/#3266 is handled, though.)
